tests/TestRegistryTest.cpp

```diff
@@ -300,3 +304,22 @@ TEST(TestRegistry, CurrentRepetitionIsCorrectTwo)
     LONGS_EQUAL(2, myRegistry->getCurrentRepetition());
 }
 
+class MyTestPluginDummy: public TestPlugin
+{
+public:
+    MyTestPluginDummy(const SimpleString& name) : TestPlugin(name) {}
+    virtual ~MyTestPluginDummy() {}
+    virtual void runAllPreTestAction(UtestShell&, TestResult&) _override {}
+    virtual void runAllPostTestAction(UtestShell&, TestResult&) _override {}
+};
+
+TEST(TestRegistry, ResetPluginsWorks)
+{
+    MyTestPluginDummy plugin1("Plugin-1");
+    MyTestPluginDummy plugin2("Plugin-2");
+    myRegistry->installPlugin(&plugin1);
+    myRegistry->installPlugin(&plugin2);
+    LONGS_EQUAL(2, myRegistry->countPlugins());
+    myRegistry->resetPlugins();
+    LONGS_EQUAL(0, myRegistry->countPlugins());
```

Comment:
I totally agree with you. Yet there are situations (such as this one) where I want to _assert_ that my Arrange is the way I think it is.

Actually, I started out installing the NullPlugin twice. Guess what? The first assertion failed, because that is used as "no plugin"... The test case would have passed, but without proving the point it was supposed to prove.

Would you consider asserting the validity of your arrangements before the Action always "wrong"? I'm not opposed to splitting this in two tests, if you prefer. If set up identically, the first test could prove the validity of my Assertion (two plugins are indeed installed)